includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2227,4 +2229,30 @@ public function get_upe_available_payment_methods() {
 			]
 		);
 	}
+
+	/**
+	 * Enable UPE feature flag. CTA from Admin Notes. See WC_Payments_Notes_Additional_Payment_Methods.
+	 */
+	public function maybe_enable_upe_feature_flag() {
+		$nonce = '';
+		if ( ! empty( $_GET['_wpnonce'] ) ) {
+			$nonce = sanitize_key( $_GET['_wpnonce'] );
+		}
+
+		if (
+			! wp_verify_nonce( $nonce, 'enable-upe' ) ||
+			empty( $_GET['page'] ) ||
+			'wc-settings' !== $_GET['page'] ||
+			empty( $_GET['action'] ) ||
+			'enable-upe' !== $_GET['action']
+		) {
+			return;
+		}
+
+		// Enable UPE.
+		WC_Payments_Features::update_upe_enabled( true );
+
+		$wcpay_settings_url = admin_url( 'admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments' );
+		wp_safe_redirect( $wcpay_settings_url );
```

Comment:
Double checked https://developer.wordpress.org/reference/functions/wp_safe_redirect/, good call out!

One note, `exit;` will exit the unit test that I put in. Depending on how it will look like after some changes, the test may not be needed anymore and I will remove it.

Reply:
Gotcha, that makes sense - maybe `wp_die` could have a different behavior 🤷 not sure tho 😄